source/module_elecstate/module_charge/charge.h

```diff
@@ -81,19 +80,23 @@ class Charge
 
     void rho_mpi(void);
 
-	// mohan add 2021-02-20
-	int nrxx; // number of r vectors in this processor
-	int ngmc; // number of g vectors in this processor
-	int nspin; // number of spins
+    void write_dipole(const double *rho_save,
```

Comment:
Don't add declaration in charge.h for a function in module_io, please treat Charge class as a input parameter of this function and move the declaration to module_io.